assets/app/scripts/controllers/create/nextSteps.js

```diff
@@ -0,0 +1,79 @@
+"use strict";
+/* jshint unused: false */
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:CreateFinishedController
+ * @description
+ * # CreateFinishedController
+ * Controller of the openshiftConsole
+ */
+angular.module("openshiftConsole")
+  .controller("CreateFinishedController", function($scope, $http, $routeParams, DataService, $q, $location, TaskList, $parse, Navigate, $filter, imageObjectRefFilter, failureObjectNameFilter) {
+    var displayNameFilter = $filter('displayName');
+    var watches = [];
+
+    $scope.emptyMessage = "Loading...";
+    $scope.alerts = [];
+    $scope.baseUrl = location.origin;
+    $scope.projectName = $routeParams.project;
+    $scope.unfilteredBuildConfigs = {};
+    $scope.projectPromise = $.Deferred();
+    DataService.get("projects", $scope.projectName, $scope).then(function(project) {
+      $scope.project = project;
+      $scope.projectPromise.resolve(project);
+    });
+
+    var name = $routeParams.name;
+    if (name) {
+      $scope.name = name;
+
+      watches.push(DataService.watch("buildconfigs", $scope, function(buildconfigs) {
```

Comment:
There is, I plumbed the JS API for it in data.js in my PR, but its not yet
actually doing the scoped down api call.

But I believe the watch api will still 404 if the resource doesn't exist
yet.  @smarterclayton is there any way to watch on a resource type and have
it filter down to a particular resource when that resource doesnt yet
exist?  Other than possibly using a labelfilter if we know what labels were
applied to the BC.

On Thu, Sep 24, 2015 at 3:37 PM, Fabiano Franz <notifications@github.com>
wrote:

> In assets/app/scripts/controllers/create/nextSteps.js
> <https://github.com/openshift/origin/pull/4728#discussion_r40362604>:
>
> > +    $scope.emptyMessage = "Loading...";
> > +    $scope.alerts = [];
> > +    $scope.baseUrl = location.origin;
> > +    $scope.projectName = $routeParams.project;
> > +    $scope.unfilteredBuildConfigs = {};
> > +    $scope.projectPromise = $.Deferred();
> > +    DataService.get("projects", $scope.projectName, $scope).then(function(project) {
> > +      $scope.project = project;
> > +      $scope.projectPromise.resolve(project);
> > +    });
> > +
> > +    var name = $routeParams.name;
> > +    if (name) {
> > +      $scope.name = name;
> > +
> > +      watches.push(DataService.watch("buildconfigs", $scope, function(buildconfigs) {
>
> Got it. Yeah, mostly if we just get when the page loads, the bc may not
> be there yet. That's why I watch. Is there such a thing like providing an
> id/name as context to watch?
>
> On the other hand, I think I could also wait until all tasks are completed
> and fire the get after that. Let me check if it works.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4728/files#r40362604>.
>